docs/onboarding-guide.md

```diff
@@ -0,0 +1,171 @@
+    * See [here](https://aws.github.io/aws-dotnet-deploy/docs/support/) to look at the support matrix between recipes and different .NET application types.
+* [**DeploymentBundleDefinitions**](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes/DeploymentBundleDefinitions) - It contains the common option settings that users can customize for container based and non-container based recipes. All recipes have a [DeploymentBundle](https://github.com/aws/aws-dotnet-deploy/blob/66fb1e34bbea10bbd995596e1eec801b79ff9102/src/AWS.Deploy.Recipes/RecipeDefinitions/aws-deploy-recipe-schema.json#L59) property that specifies how to package the user’s .NET project for deployment. 
+    * *Container* - Indicates that Docker is used to build the Docker image and pushed to Amazon Elastic Container Registry.
+    * *DotnetPublishZipFile -* Indicates that the `dotnet publish`  is used to prepare the user’s .NET project for deployment.
+* [**CdkTemplates**](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes/CdkTemplates) - It contains the CDK templates for recipe files found [here](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes/RecipeDefinitions). All recipes with the [DeploymentType](https://github.com/aws/aws-dotnet-deploy/blob/66fb1e34bbea10bbd995596e1eec801b79ff9102/src/AWS.Deploy.Recipes/RecipeDefinitions/aws-deploy-recipe-schema.json#L53) set to *“CdkProject”* have their own specific CDK template identified by the [CdkProjectTemplate](https://github.com/aws/aws-dotnet-deploy/blob/66fb1e34bbea10bbd995596e1eec801b79ff9102/src/AWS.Deploy.Recipes/RecipeDefinitions/aws-deploy-recipe-schema.json#L65) property.
+
+
+
+### **[AWS.Deploy.CDK.Common](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes.CDK.Common)**
+
+This component contains the common utility classes and methods that are used by the CDK templates found [here](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes/CdkTemplates).
+
+
+### [AWS.Deploy.Common](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Common)
+
+This component contains POCO objects and common logic that is used by other components in the deployment tool
+**Note** - This project does not reference any other project except for `../src/AWS.Deploy.Constants/AWS.Deploy.Constants.projitems`
+
+#### Key Components
+
+* [**ProjectDefinitionParser**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Common/ProjectDefinitionParser.cs) - This class contains the logic to validate whether a .NET project exists at the path specified via `dotnet aws deploy —project-path <PROJECT-PATH>`. It also parses the user’s csproj/fsproj file and returns a [ProjectDefinition](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Common/ProjectDefinition.cs)
```

Comment:
This list is inconsistent on whether to put the `**` for bolding inside the brackets or outside the brackets. I'm assuming we want them on the outside but I didn't look to see how this renders.

Reply:
I created this markdown file by exporting from a quip doc. It renders correctly with `**` being inside the bracket. But for conistency, I have moved all occurrences to be outside the bracket.